src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  injure(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.alive.filter((animal) => animal.health <= 0);
```

Comment:
The `filter` method here is not being used correctly. It should be used to update `Animal.alive` by filtering out dead animals. As per the checklist, avoid using `indexOf` and `splice` in a loop for performance reasons. Instead, you should call `Animal.nature()` here to remove the dead animal from the `Animal.alive` array.